### DISCUSSION TITLE
audio "pausing" when loading acts

### DISCUSSION THREAD

**Discussion by jpsteccato** [ppooll-dev/ppooll#140] [General]
starting a little discussion here to get an understanding of the little "pause" you hear when you are loading a new audio act.  

from my understanding, this is because of the vector delay introduced by s~/r~ objects in Max.  

on discord, henk mentioned that he noticed a (possibly larger? or totally new to him, not sure...) pause when he updated to the latest version.

my hunch is that some of the fancy (cordless) audio patching changes with ll.blues and ll.r~/ll.s~ is causing some of this, and maybe the vector delay would be less noticeable if all audio sending/receiving was done with patchcords (where possible).  for example, maybe it's better to patch audio directly from the subpatch of an act to the ll.blues object itself

@klausfilip @chausch any insight here on how all this works?

**Comment by chausch** [comment]
i reckon it was introduced to allow mc channel amount changes in the acts, right @klausfilip ?

**Comment by klausfilip** [comment]
hi there, 
i am pretty sure, its not the cordless audio patching, but this can be tested/compared.
some (not all) multichannel objects need audio to be switched off-on (with defered on), to update their channel-count.
in ho_st>ho_st-subpatches>audio>ll_audio there is a receive "ll_audio" that is used by some abstraction (like ll.blues, etc) that need to update channels...

**Comment by klausfilip** [comment]
it is very obvious, that this is an ugly hack. and it might be possible that we can at least avoid this forced audio-off-on at patch loading, when the channel count has a default value.

**Comment by klausfilip** [comment]
...if i disconnect the mentioned [r ll_audio] in ho_st and load sinus for example, there is no audio hickup. so this suggests, that i have to look into ll_blues when to ask for audio-off-on...

**Comment by klausfilip** [comment]
i made 2 changes:
in ho_st: a bang to [s ll_audio] (audio_off_on) was sent every time an act was loaded. not sure why i implemented this, so we only can watch if this runs into issues.
in ll.blues: [s ll_audio] only if channels change from default 1 in, 2 out.
tested with loading "sinus" while audio is playing: no interrupt and correct behaviour when playing "sinus".

**Reply by jpsteccato** [reply]
This sounds great klaus!  i will be able to fully test on Monday, sorry if I started some commotion with all this :)

**Comment by jpsteccato** [comment]
I've done some testing and this is definitely an improvement for acts with default channel configs (1 in, 2 out).  

wondering if there is a way to set those [change] objects to be whatever the default ll.blues for the act is (ie xgroove 2 in 2 out)
